packages/block-editor/src/store/reducer.js

```diff
@@ -755,6 +755,22 @@ const withResetControlledBlocks = ( reducer ) => ( state, action ) => {
 	return reducer( state, action );
 };
 
+/**
+ * Adds the selected pattern category to the outer block.
+ *
+ * @param {Object} action        Dispatched action.
+ * @param {Object} action.meta   The meta object from the action.
+ * @param {Array}  action.blocks The array of blocks to insert.
+ *
+ * @return {Array} The blocks array.
+ */
+function addSelectedPatternCategoryToOuterBlock( { meta, blocks } ) {
+	if ( meta?.category && blocks.length === 1 ) {
+		blocks[ 0 ].attributes.metadata.categories = [ meta.category.name ];
```

Comment:
I'm wondering if `category` is too generic for the metadata. It's more about the "pattern category at point of insertion".